src/otx/algo/classification/torchvision_model.py

```diff
@@ -155,45 +154,148 @@ def __init__(
             optimize_gap=True,
         )
 
+    def _get_head(self, net: nn.Module) -> nn.Module:
+        """Returns the head of the model."""
+        head: nn.Module
+        last_layer = list(net.children())[-1]
+        layers = []
+        if self.task == "multiclass":
+            classifier_len = len(list(last_layer.children()))
+            if classifier_len >= 1:
+                feature_channel = list(last_layer.children())[-1].in_features
+                layers = list(last_layer.children())[:-1]
+                self.use_layer_norm_2d = layers[0].__class__.__name__ == "LayerNorm2d"
+
+                head = nn.Sequential(*layers, nn.Linear(feature_channel, self.num_classes))
+            else:
+                feature_channel = last_layer.in_features
+                head = nn.Linear(feature_channel, self.num_classes)
+
+            if self.train_type == "semi_supervised":
+                self.neck = nn.Sequential(*layers) if layers else None
+                head = OTXSemiSLLinearClsHead(
+                    num_classes=self.num_classes,
+                    in_channels=feature_channel,
+                    loss=self.loss_module,
+                )
```

Comment:
If self.train_type == "semi_supervised", `head` seems to be assigned twice.
How about updating if-else statement?

Reply:
Thanks, Fixed - https://github.com/openvinotoolkit/training_extensions/pull/3614/commits/6bc9bfa48aa11b0f5b4677f2e6a489e830ceda75